### ISSUE TITLE
Link should ignore instances and namespaces created to host external objects

### ISSUE THREAD

**Issue by mkromberg** [Dyalog/link#397]
Link should ignore:

- Instances (9.2)
- External Classes (9.6) and Interfaces 9.7
- Namespaces created to host external classes and interfaces (e.g. "System" when using .NET)

**Comment by aplteam** [comment]
Also, instances of Forms created with `⎕WC` should be ignored as well when `KeepOnClose` is `0` but kept with `KeepOnClose←1`

**Comment by mkromberg** [comment]
I think all instances of forms must ALWAYS be ignored, they are instances and we don't have a mechanism to serialise and restore them.

**Comment by aplteam** [comment]
`KeepOnClose` is a mechanism that allows to store code in a namespace that, one the Form is created, happens to have the same name as that Form without being the same thing. Once the Form is closed the namespace remains with the code as if it were an ordinary namespace.

For that reason Link should and can without difficulties handle the code in anything that has `KeepOnClose←1`.

**Comment by abrudz** [comment]
> For that reason Link should and can without difficulties handle the code in anything that has `KeepOnClose←1`.

I agree, assuming the form has indeed been closed. However, that changes its nameclass to 9.1, so it would already be handled:
```apl
      'f'⎕WC'Form'('KeepOnClose' 1)
      ⎕NC⊂,'f'
9.2
      2⎕NQ f 'Close'
      ⎕NC⊂,'f'
9.1
```

**Comment by aplteam** [comment]
The workflow is usually the other way around:

1. You bring in code with Link into `Foo` which is a 9.1 namespace
2. You create a Form with `'Foo' ⎕WC'Form' ('KeepOnClose'  1`)
3. The namespace and the Form exist in parallel under the same name, but the name class is now that of an instance of a GUI

At this point when code within `Foo` is changed via the editor, that change needs to be tracked by Link, despite the Form being not closed.

Once the Form is closed it's a 9.1 namespace anyway.

**Comment by mkromberg** [comment]
Fixed in 3.0.10
